functions/lessonNotifications.js

```diff
@@ -0,0 +1,153 @@
+const { getFirestore } = require("firebase-admin/firestore");
+const { sendNotification, preparePayload } = require("./notification");
+
+
+const db = getFirestore();
+
+module.exports = {
```

Comment:
Each function has a clearly defined responsibility. This separation of concerns makes the code more readable and maintainable.